engine/src/main/java/org/terasology/rendering/nui/layers/ingame/inventory/InventoryGrid.java

```diff
@@ -52,14 +52,14 @@
     public void update(float delta) {
         super.update(delta);
 
-        int numSlots = InventoryUtils.getSlotCount(getTargetEntity()) - getCellOffset();
+        int numSlots = getNumSlots();
 
         // allow the UI to shrink the cell count if the inventory shrinks
         if (numSlots < cells.size()) {
-            cells.clear();
-        }
-
-        if (numSlots > cells.size() && cells.size() < getMaxCellCount()) {
+            for(int i = cells.size()-1; i >= 0; --i) {
```

Comment:
Can you run Checkstyle and address any issues it reports please? I can see a couple of minor style issues (mostly missing spaces).